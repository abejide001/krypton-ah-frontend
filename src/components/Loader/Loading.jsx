import React from 'react';
import { Loader } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import './Loading.scss';

const Loading = ({ size }) => <Loader size={size} active inline="centered" />;

export default Loading;

Loading.propTypes = {
  size: PropTypes.string
};

Loading.defaultProps = {
  size: 'tiny'
};
