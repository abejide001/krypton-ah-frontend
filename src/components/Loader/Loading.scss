@import '../../styles/partials/colors';
.ui.dimmer .ui.loader:after {
  border-color: $black transparent transparent;
}

