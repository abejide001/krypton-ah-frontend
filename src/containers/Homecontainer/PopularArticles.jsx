import React from 'react';
import { Grid, Image, Card } from 'semantic-ui-react';
import './HomeContainer.scss';
import business from './images/business.png';
import Title from '../../components/sectionHeader';
import { popularArticles } from '../../constants';

export default function PopularArticles() {
  return (
    <div>
      <Title text="Popular" />
      <Grid columns={2} divided>
        <Grid.Row>
          <Grid container className="reveal" columns={2}>
            {popularArticles.map(popularArticle => (
              <Grid.Column>
                <Card className="description">
                  <p className="Popular title">{popularArticle.title}</p>
                  {popularArticle.description}
                </Card>
                <Image src={business} />
              </Grid.Column>
            ))}
          </Grid>
        </Grid.Row>
      </Grid>
    </div>
  );
}
