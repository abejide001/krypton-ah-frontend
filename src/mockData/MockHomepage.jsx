import React from 'react';
import { Link } from 'react-router-dom';

const selectedArticleId = '65719288-0395-445e-b587-2b98b70bdec9';

export default function Homepage() {
  return (
    <Link to={`article/${selectedArticleId}`}>
      <div>
        <button type="button">Read this article</button>
      </div>
    </Link>
  );
}
