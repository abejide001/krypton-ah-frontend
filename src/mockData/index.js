import { API_BASE_URL } from '../constants';

const items = [
  {
    description:
      'The Touch Bar replaces the function keys that have long occupied the top of your keyboard with something much more versatile and capable.16 It changes automatically based on what youâ€™re doing to show you relevant The Touch Bar replaces the function keys that have long occupied the top of your keyboard with something much more versatile and capable.16 It changes automatically based',
    shortdescription:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach, while an optimist drains the barrel, fishes out all the apples and makes pie',
    shorterdescription:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Popular Title'
  }
];

const categories = ['Politics', 'Family', 'Techonology', 'Health', 'Sports', 'Wellbeing'];

const popularArticles = [
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Popular Title'
  },
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Popular Title'
  },
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Popular Title'
  },
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Popular Title'
  }
];

const mockHeader = 'this is a placeholder header';
const trendingArticles = [
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Trending Title'
  },
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Trending Title'
  },
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Trending Title'
  },
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Popular Title'
  },
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Popular Title'
  }
];

const payload = {
  firstname: 'john',
  lastname: 'joseph',
  email: 'jo@jos.com',
  password: 'password123R'
};
const fakeUser2 = {
  email: 'jo@jos.com',
  password: 'password'
};
const signupEndpoint = `${API_BASE_URL}/api/v1/users/signup`;
const signinEndpoint = `${API_BASE_URL}/api/v1/users/signin`;

const signupOkResponse = {
  status: 201,
  success: true,
  data: {
    message:
      'Account successfully created, Kindly check your email to activate your account. In case you did not receive the activation link in your mail, kindly visit this link localhost:3000/api/v1/users/resend/activation/mail to resend the mail.'
  }
};

const mockResponse = {
  message: 'Successfully logged in'
};
const error = {
  response: {
    status: 400,
    success: false,
    data: {
      message: 'Oops! something went wrong, kindly try again'
    }
  }
};
const loginOkResponse = {
  status: 200,
  success: true,
  data: {
    token: 'sampleToken',
    message: 'You have sccessfully login'
  }
};
const loginBadResponse = {
  status: 400,
  success: false,
  data: {
    message: 'Incorrect credentials'
  }
};

const startAction = [{ type: 'START' }];

const signupSuccessAction = [
  {
    type: 'SIGNUP_SUCCESS',
    payload
  }
];
const signupFailureAction = [
  {
    type: 'SIGNUP_FAILURE',
    payload
  }
];

const socialToken = 'hyakfhkdhiuashvk98924cxndusjdusnfixomj9';

const mockPath = 'auth/facebook/';
const profileData = {
  firstname: 'Isaiah',
  lastname: 'Afolayan',
  profileImage: '',
  bio: 'Software developer at Andela',
  id: 'd1564aa6-e81d-41ab-8469-0aa573f4a6c5',
  country: 'Nigeria',
  phonenumber: '+2348145692034',
  gender: 'male',
  username: 'Isaiah'
};

const getUserFailureResponse = {
  data: {
    success: false,
    message: 'error retieving user'
  }
};

const getUserSuccessResponse = {
  data: {
    success: true,
    message: 'users retrieved succesfully',
    data: {
      id: 'd1564aa6-e81d-41ab-8469-0aa573f4a6c5',
      email: '1skyboat1@gmail.com',
      password: '$2b$08$Q5LSk9wYgQeuxtSTGeq/0eBMclMVg1B7wE/hfWRCjsUb9YHkgx4h.',
      firstname: 'firstname',
      lastname: 'lastname',
      isverified: true,
      createdAt: '2019-02-08T09:58:13.163Z',
      updatedAt: '2019-02-08T09:58:13.163Z',
      userprofile: {
        id: 1,
        avatar: null,
        bio: 'A world-class software developer',
        UserId: 'd1564aa6-e81d-41ab-8469-0aa573f4a6c5',
        username: 'eltneg',
        phonenumber: '+2348137457862',
        country: 'Nigeria',
        gender: 'male',
        emailnotification: false,
        createdAt: '2019-02-14T16:54:58.292Z',
        updatedAt: '2019-02-18T08:41:31.199Z'
      },
      articles: [
        {
          title: 'Article title6',
          description: 'valid article description',
          isPublished: false,
          createdAt: '2019-02-14T17:53:01.904Z',
          id: 'a76d2040-7510-43d8-b58b-a32aba67ab48'
        }
      ]
    }
  }
};

const jwtToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlMb2FkIjoiZDE1NjRhYTYtZTgxZC00MWFiLTg0NjktMGFhNTczZjRhNmM1IiwiaWF0IjoxNTUwNDY1MjY0LCJleHAiOjE1NTA1NTE2NjR9.kFe-uZW0XSQCV_gb1WVV3Z3oCxZffJAOOaKXSj0w0m0';

export {
  payload,
  fakeUser2,
  signupEndpoint,
  signinEndpoint,
  mockResponse,
  signupOkResponse,
  startAction,
  signupSuccessAction,
  signupFailureAction,
  error,
  loginOkResponse,
  loginBadResponse,
  items,
  categories,
  popularArticles,
  trendingArticles,
  mockHeader,
  socialToken,
  mockPath,
  profileData,
  jwtToken,
  getUserSuccessResponse,
  getUserFailureResponse
};
