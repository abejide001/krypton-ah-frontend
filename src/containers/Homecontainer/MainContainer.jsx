/* eslint-disable no-shadow */
import React, { Component } from 'react';
import { Grid } from 'semantic-ui-react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Carousel from '../../components/carousel/index';
import Sidewidget from '../sideWidgetContainer';
import FeaturedArticles from './featuredArticles';
import PopularArticles from './popular';
import './HomeContainer.scss';
import {
  fetchpopular,
  fetchtrending,
  fetchfeatured
} from '../../actions/fetchArticlesAction/fetchArticlesActions';
import Title from '../../components/sectionHeader';
import loader from '../loaders/responsechecker';

class MainContainer extends Component {
  componentWillMount() {
    const { fetchpopular, fetchtrending, fetchfeatured } = this.props;
    fetchpopular();
    fetchtrending();
    fetchfeatured();
  }

  render() {
    const { articles } = this.props;
    return (
      <div className="main">
        <Grid className="content" divided="vertically">
          <Grid.Row columns={1}>
            <Grid.Column>
              {loader(
                articles.featuredArticlesResponsefailure,
                <Carousel featured={articles.featuredArticlesResponse} />
              )}
            </Grid.Column>
            <Grid.Column className="second_grid">
              <Title text="featured" />
              {loader(
                articles.featuredArticlesResponsefailure,
                <FeaturedArticles articles={articles.featuredArticlesResponse} />
              )}
              <Title text="Popular" />
              {loader(
                articles.popularArticlesResponsefailure,
                <PopularArticles popular={articles.popularArticlesResponse} />
              )}
            </Grid.Column>
          </Grid.Row>
        </Grid>
        <Grid className="side" divided="vertically">
          <Sidewidget />
        </Grid>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  articles: state.articles
});

export default connect(
  mapStateToProps,
  { fetchpopular, fetchtrending, fetchfeatured }
)(MainContainer);

MainContainer.defaultProps = {
  articles: [],
  fetchpopular: [],
  fetchfeatured: [],
  fetchtrending: []
};

MainContainer.propTypes = {
  articles: PropTypes.func,
  fetchpopular: PropTypes.func,
  fetchfeatured: PropTypes.func,
  fetchtrending: PropTypes.func
};
