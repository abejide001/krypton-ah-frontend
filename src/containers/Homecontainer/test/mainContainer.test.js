import React from 'react';
import { shallow } from 'enzyme';
import { MainPage } from '../MainContainer';
import PopularArticles from '../popular';
import FeaturedArticles from '../featuredArticles';
import Loader from '../../loaders/componentLoader';
import {
  fetchpopular,
  fetchfeatured
} from '../../../actions/fetchArticlesAction/fetchArticlesActions';
import { defaultprop } from '../../../mockData';

describe('home container test', () => {
  it('should test for the home component', () => {
    const wrapper = shallow(<MainPage fetchpopular={fetchpopular} fetchfeatured={fetchfeatured} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should test for the popular articles component', () => {
    const wrapper = shallow(<PopularArticles popular={false} />);
    expect(wrapper.find('Grid')).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('should test for the popular articles component', () => {
    const wrapper = shallow(<PopularArticles popular={defaultprop.data} />);
    expect(wrapper.find('Grid')).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('should test for the featuredarticles component', () => {
    const wrapper = shallow(<FeaturedArticles articles={defaultprop} />);
    expect(wrapper.find('Grid')).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('should test for the featuredarticles component', () => {
    const wrapper = shallow(<FeaturedArticles />);
    expect(wrapper.find('Grid')).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('should test for the featuredarticles component', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });
});
