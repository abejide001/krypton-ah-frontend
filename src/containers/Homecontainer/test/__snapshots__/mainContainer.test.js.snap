// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home container test should test for the featuredarticles component 1`] = `
<Fragment>
  <sectionHeader
    text="featured"
  />
  <Grid
    columns={2}
    divided={true}
  >
    <GridRow>
      <GridColumn
        className="rightGrid"
      >
        <Card
          header="Elliot Baker"
          image="test-file-stub"
          items={
            Array [
              Object {
                "content": "Apple's news at CES 2019 shows it's facing some hard truths",
                "description": "The Touch Bar replaces the function keys that have long occupied the top of your keyboard with something much more versatile and capable.16 It changes automatically based on what you’re doing to show you relevant The Touch Bar replaces the function keys that have long occupied the top of your keyboard with something much more versatile and capable.16 It changes automatically based",
                "shortdescription": "When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach, while an optimist drains the barrel, fishes out all the apples and makes pie",
                "shorterdescription": "When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ",
                "title": "Popular Title",
              },
            ]
          }
        />
        <Card
          className="description"
        >
          The Touch Bar replaces the function keys that have long occupied the top of your keyboard with something much more versatile and capable.16 It changes automatically based on what you’re doing to show you relevant The Touch Bar replaces the function keys that have long occupied the top of your keyboard with something much more versatile and capable.16 It changes automatically based
          <Icons
            numberofcomments={23}
            ratingIcon={null}
          />
        </Card>
      </GridColumn>
      <GridColumn>
        <ArticlesFeedContainer
          articles={
            Array [
              Object {
                "content": "Apple's news at CES 2019 shows it's facing some hard truths",
                "description": "When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach",
                "title": "Popular Title",
              },
              Object {
                "content": "Apple's news at CES 2019 shows it's facing some hard truths",
                "description": "When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach",
                "title": "Popular Title",
              },
              Object {
                "content": "Apple's news at CES 2019 shows it's facing some hard truths",
                "description": "When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach",
                "title": "Popular Title",
              },
              Object {
                "content": "Apple's news at CES 2019 shows it's facing some hard truths",
                "description": "When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach",
                "title": "Popular Title",
              },
            ]
          }
          className="featured-widget"
        />
      </GridColumn>
    </GridRow>
  </Grid>
</Fragment>
`;

exports[`home container test should test for the home component 1`] = `
<div
  className="main"
>
  <Grid
    className="content"
    divided="vertically"
  >
    <GridRow
      columns={1}
    >
      <GridColumn>
        <carousel
          author="Placeholder author name"
          featuredImage="https://res.cloudinary.com/ah-krypton/image/upload/v1550007680/MainImage.png"
          title="Placeholder Article Title"
        />
      </GridColumn>
      <GridColumn
        className="second_grid"
      >
        <FeaturedArticles />
        <PopularArticles />
      </GridColumn>
    </GridRow>
  </Grid>
  <Grid
    className="side"
    divided="vertically"
  >
    <WidgetContainer />
  </Grid>
</div>
`;

exports[`home container test should test for the popular articles component 1`] = `
<Fragment>
  <sectionHeader
    text="Popular"
  />
  <Grid
    columns={2}
    divided={true}
  >
    <GridRow>
      <Grid
        className="reveal"
        columns={2}
        container={true}
      >
        <GridColumn
          key="Popular Title"
        >
          <Card
            className="description"
          >
            <p
              className="popular title"
            >
              Popular Title
            </p>
            When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach
          </Card>
          <Image
            as="img"
            src="test-file-stub"
            ui={true}
          />
        </GridColumn>
        <GridColumn
          key="Popular Title"
        >
          <Card
            className="description"
          >
            <p
              className="popular title"
            >
              Popular Title
            </p>
            When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach
          </Card>
          <Image
            as="img"
            src="test-file-stub"
            ui={true}
          />
        </GridColumn>
        <GridColumn
          key="Popular Title"
        >
          <Card
            className="description"
          >
            <p
              className="popular title"
            >
              Popular Title
            </p>
            When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach
          </Card>
          <Image
            as="img"
            src="test-file-stub"
            ui={true}
          />
        </GridColumn>
        <GridColumn
          key="Popular Title"
        >
          <Card
            className="description"
          >
            <p
              className="popular title"
            >
              Popular Title
            </p>
            When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach
          </Card>
          <Image
            as="img"
            src="test-file-stub"
            ui={true}
          />
        </GridColumn>
      </Grid>
    </GridRow>
  </Grid>
</Fragment>
`;
